getpaid/backends/payu/processor.py

```diff
@@ -42,92 +53,81 @@ class PaymentProcessor(BaseProcessor):
-            "unit_price": "unitPrice",
-            "first_name": "firstName",
-            "last_name": "lastName",
-            "order_id": "extOrderId",
-            "customer_ip": "customerIp",
-            "notify_url": "notifyUrl",
-        }
-
-        raw_items = self.payment.get_items()
-
         context = {
-            "order_id": self.payment.get_unique_id(),
-            "customer_ip": loc if not request else request.META.get("REMOTE_ADDR", loc),
+            "notify_url": self.get_notify_url(),
+            "continue_url": self.get_continue_url(),
+            "customer_ip": self.get_customer_ip(request),
             "description": self.payment.description,
             "currency": self.payment.currency,
-            "amount": self.payment.amount_required,
-            "buyer": self.payment.get_buyer_info()
+            "amount": self.get_total_amount(),
+            "order_id": self.payment.get_unique_id(),
+            "buyer": self.payment.get_buyer_info(),
         }
 
         if self.get_setting("is_marketplace", False):
-            shopping_carts = []
-            for shopping_cart in raw_items:
-                products = [
-                    {key_trans.get(k, k): v for k, v in product.items()}
-                    for product in shopping_cart["products"]
-                ]
-                shopping_carts.append({
-                    **shopping_cart,
-                    "products": products
-                })
-            context["shopping_carts"] = shopping_carts
+            context["shopping_carts"] = self.get_shopping_carts()
         else:
-            products = [
-                {key_trans.get(k, k): v for k, v in product.items()}
-                for product in raw_items
-            ]
-            context["products"] = products
+            context["products"] = self.get_products()
 
-        if self.get_setting("confirmation_method", self.confirmation_method) == "PUSH":
-            context["notify_url"] = urljoin(
-                our_baseurl, reverse("getpaid:callback", kwargs={"pk": self.payment.pk})
-            )
-        if camelize_keys:
-            return {key_trans.get(k, k): v for k, v in context.items()}
         return context
 
-    def get_paywall_method(self):
-        return self.get_setting("paywall_method", self.method)
+    def get_notify_url(self):
+        backend_url = settings.GETPAID_BACKEND_HOST
+        return urljoin(
+            backend_url, reverse(self.callback_url_name, kwargs={"pk": self.payment.pk})
+        )
+
+    def get_continue_url(self):
+        frontend_url = self.get_setting("frontend_host")
+        return self.get_setting("continue_url").format(
+            frontend_url=frontend_url, payment_id=self.payment.id
+        )
+
+    def get_customer_ip(self, request=None):
+        customer_ip = "127.0.0.1"
+        if request:
+            customer_ip = request.META.get("REMOTE_ADDR", customer_ip)
+        return customer_ip
+
+    def get_total_amount(self):
```

Comment:
get_total_amount to powinien być Decimal, proponuję wyrzucić w ogóle tę metodę

zwłaszcza, że niepotrzebnie mnoży ona kwotę przez 100